code/modules/projectiles/gun_helpers.dm

```diff
@@ -533,7 +537,7 @@ should be alright.
 
 	usr.visible_message("<span class='notice'>[usr] strips [A] from [src].</span>",
 	"<span class='notice'>You strip [A] from [src].</span>", null, 4)
-	A.Detach(usr)
+	A.detach_from_master_gun(usr)
```

Comment:
guns have gun_user since the rework, you could use that rather than usr. (gun user is null only when the gun is not in a hand)

Reply:
this is in a verb